schema/gendummydata.py

```diff
@@ -18,6 +18,18 @@
 
 import bcrypt
 
+def getosname():
+    if (os.name == "posix"):
+        try:
+            with open("/etc/os-release", "r") as f:
+                for lines in f:
+                    if(lines.find("NAME") > -1):
+                        osname = lines.split('=')[1].strip()
```

Comment:
Running this part on my local machine is returning `focal` for `osname`, presumably because of this at the end of my `/etc/os-release` file:

```
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

It should be checking that `NAME` variable at the top of the file, right?

---

I'm not sure how much we can rely on it as I don't think it's installed by default (and I'm not sure if you've thought about it yet either), but it looks like we can get the OS name with the `lsb_release -is` command.

Reply:
yupp, have commented the code needs to be tested for ubuntu, now we know it won't work properly for ubuntu. The problem with lsb_release is, I am not sure if it needs the package installed. Where as the os-release is bound to be present in any and all linux standard distros.

Reply:
One Ideal method would be to have an exported environment variable as well in case lsb_release or os-release doesn't exists then the dev can just export the OSNAME and force the code to take the data from env var instead.